sklearn/utils/estimator_checks.py

```diff
@@ -3933,7 +3933,7 @@ def _enforce_estimator_tags_y(estimator, y):
         and not tags.classifier_tags.multi_class
         and y.size > 0
     ):
-        y = np.where(y == y.flat[0], y, y.flat[0] + 1)
+        y = np.where(y == y.min(), y, y.min() + 1)
```

Comment:
Why don't we just put 0 and 1 ? or -1 and 1 ?
Something like `y = np.where(y == y.min(), 0, 1)`

Reply:
It could be in conflict with enforcing `tags.target_tags.positive_only`  (although a binary classifier with positive target will probably never exist...)

Testing locally it breaks `check_supervised_y_no_nan`, because this test assumes  that `_enforces_estimator_tags_y` keeps `np.inf` and `np.nan` in the target (and checks that the estimator then properly fails).

Reply:
Alright, let's keep it then :)